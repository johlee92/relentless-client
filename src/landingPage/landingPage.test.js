import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingPage from './landingPage';
import { BrowserRouter } from 'react-router-dom';

describe(`LandingPage component`, () => {
    const props = {
        className: 'test-class-name'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <LandingPage />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component by default', () => {
        const wrapper = shallow(<LandingPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the component from props', () => {
        const wrapper = shallow(<LandingPage {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
