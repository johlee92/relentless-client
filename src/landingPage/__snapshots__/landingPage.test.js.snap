// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders the component by default 1`] = `
<div>
  <PageNav />
  <div
    className="landing-page"
  >
    <div
      className="landing-page-container"
    >
      <div>
        <h2>
          Live with Intent. Have clarity. Be hyperfocused. Be disciplined. And Achieve your goals.
        </h2>
      </div>
      <img
        alt="track and field lanes into the unknown"
        className="landing-page-image"
        src="trackField.jpeg"
      />
      <Link
        to="/user1"
      >
        <button
          className="try-for-free"
        >
          Try for Free
        </button>
      </Link>
      <div>
        <h2>
          Our Three Pillars
        </h2>
      </div>
      <div>
        We strive to help you achieve your goals by focusing on three factors.
      </div>
      <div
        className="methodology"
      >
        <div
          className="methodology-cards"
          name="clarity"
          onClick={[Function]}
        >
          <h3
            name="clarity"
            onClick={[Function]}
          >
            Clarity
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Know the destinations (i.e. setting the goals) is an important step of the process
          </p>
        </div>
        <div
          className="methodology-cards"
          name="hyperfocus"
          onClick={[Function]}
        >
          <h3
            name="hyperfocus"
            onClick={[Function]}
          >
            Hyperfocus
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Determine which three goals are the top priorities right now to focus our precious time and energy
          </p>
        </div>
        <div
          className="methodology-cards"
          name="discipline"
          onClick={[Function]}
        >
          <h3
            name="discipline"
            onClick={[Function]}
          >
            Discipline
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Transformation requires presistent effort - breaking goals down into actionable pieces drives change
          </p>
        </div>
      </div>
      <div>
        <h2>
          Let's reach new heights together!
        </h2>
      </div>
      <img
        alt="Relentless Logo"
        className="landing-page-image"
        src="new-heights.jpeg"
      />
      <Link
        to="/user1"
      >
        <button
          className="try-for-free"
        >
          Try for Free
        </button>
      </Link>
    </div>
  </div>
  <PageFooter />
</div>
`;

exports[`LandingPage component renders the component from props 1`] = `
<div>
  <PageNav />
  <div
    className="landing-page"
  >
    <div
      className="landing-page-container"
    >
      <div>
        <h2>
          Live with Intent. Have clarity. Be hyperfocused. Be disciplined. And Achieve your goals.
        </h2>
      </div>
      <img
        alt="track and field lanes into the unknown"
        className="landing-page-image"
        src="trackField.jpeg"
      />
      <Link
        to="/user1"
      >
        <button
          className="try-for-free"
        >
          Try for Free
        </button>
      </Link>
      <div>
        <h2>
          Our Three Pillars
        </h2>
      </div>
      <div>
        We strive to help you achieve your goals by focusing on three factors.
      </div>
      <div
        className="methodology"
      >
        <div
          className="methodology-cards"
          name="clarity"
          onClick={[Function]}
        >
          <h3
            name="clarity"
            onClick={[Function]}
          >
            Clarity
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Know the destinations (i.e. setting the goals) is an important step of the process
          </p>
        </div>
        <div
          className="methodology-cards"
          name="hyperfocus"
          onClick={[Function]}
        >
          <h3
            name="hyperfocus"
            onClick={[Function]}
          >
            Hyperfocus
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Determine which three goals are the top priorities right now to focus our precious time and energy
          </p>
        </div>
        <div
          className="methodology-cards"
          name="discipline"
          onClick={[Function]}
        >
          <h3
            name="discipline"
            onClick={[Function]}
          >
            Discipline
          </h3>
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Transformation requires presistent effort - breaking goals down into actionable pieces drives change
          </p>
        </div>
      </div>
      <div>
        <h2>
          Let's reach new heights together!
        </h2>
      </div>
      <img
        alt="Relentless Logo"
        className="landing-page-image"
        src="new-heights.jpeg"
      />
      <Link
        to="/user1"
      >
        <button
          className="try-for-free"
        >
          Try for Free
        </button>
      </Link>
    </div>
  </div>
  <PageFooter />
</div>
`;
